assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -60,14 +60,24 @@ export function registerDefaults( notificationsAPI ) {
 			VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
 			VIEW_CONTEXT_SETTINGS,
 		],
-		checkRequirements: ( { select } ) => {
+		checkRequirements: async ( { select, resolveSelect } ) => {
+			await Promise.all( [
+				// The getSetupErrorMessage selector relies on the resolution
+				// of the getSiteInfo() resolver.
+				resolveSelect( CORE_SITE ).getSiteInfo(),
+				// The isAuthenticated(), hasScope() and getUnsatisfiedScopes() selectors
+				// rely on the resolution of getAuthentication().
+				resolveSelect( CORE_USER ).getAuthentication(),
+			] );
+
 			const setupErrorMessage =
 				select( CORE_SITE ).getSetupErrorMessage();
 
 			const isAuthenticated = select( CORE_USER ).isAuthenticated();
 
-			const ga4ModuleConnected =
-				select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+			const ga4ModuleConnected = await resolveSelect(
+				CORE_MODULES
+			).isModuleConnected( 'analytics-4' );
```

Comment:
We can add `resolveSelect( CORE_MODULES ).getModules()` to the `Promise.all` call and then use simple select here:
```js
	await Promise.all( [
		resolveSelect( CORE_MODULES ).getModules(),
		...
	] );
	
	...
	
	const ga4ModuleConnected = select( CORE_MODULES )
		.isModuleConnected( 'analytics-4' );
```

Reply:
I did think of this approach as well but thought it would add a lot of duplication. But it does make sense for efficiency.

Reply:
`isModuleConnected` does have its own resolver which does call `getModules()`. I'll use the `isModuleConnected`  resolver itself to avoid adding a comment since there's a direct mapping of the resolution to the select.